### PR TITLE
optical flow test fix

### PR DESCRIPTION
**Pull request by szalpal**
Signed-off-by: Michał Szołucha <mszolucha@nvidia.com>

### CODE REVIEW

File: dali/test/python/test_optical_flow.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 batch_size = 1
 sequence_length = 2
 dali_extra_path = os.environ['DALI_EXTRA_PATH']
-image_dir= os.path.join(dali_extra_path, "/db/optical_flow/slow_preset/two_frames")
+image_dir = dali_extra_path + "/db/optical_flow/slow_preset/two_frames/"
```
Comment (JanuszL) [inline dali/test/python/test_optical_flow.py]: Why?

Comment (szalpal) [reply dali/test/python/test_optical_flow.py]: IMHO `os.path.join()` is evil.
This:
`image_dir= os.path.join(dali_extra_path, "/db/optical_flow/slow_preset/two_frames")` is resolved to `/db/optical_flow/slow_preset/two_frames` cause of the `/` at the beginning.

On top of that, it doesn't make any sense using `os.path.join()` with hardcoded strings

File: dali/test/python/test_optical_flow.py [hunk 2]
```diff
@@ -23,13 +23,16 @@ def __init__(self, batch_size, num_threads, device_id):
...
     pipe = OFPipeline(batch_size=batch_size, num_threads=1, device_id=0)
     pipe.build()
     pipe_out = pipe.run()
     frames = pipe_out[0].as_cpu().as_array()
-    myarray=np.loadtxt(image_dir+'decoded_flow_vector.dat')
-    assert (0.9 < np.mean(np.abs(frames[0][0].flatten()-myarray)))
+    myarray = np.loadtxt(image_dir + '/../decoded_flow_vector.dat')
+    assert (0.9 > np.mean(np.abs(frames[0][0].flatten() - myarray)))
+
+if __name__ == '__main__':
```
Comment (klecki) [inline dali/test/python/test_optical_flow.py]: We use nosetests to run it, I suppose you don't need main for that?

File: dali/test/python/test_optical_flow.py (hunk 1 above)
Comment (JanuszL) [reply dali/test/python/test_optical_flow.py]: So you can use:
```
os.path.join(dali_extra_path, "db", "optical_flow", "slow_preset", "two_frames")
```
I would prefer portable solution than hardcodes, especially with zero effort as in this case.

Comment (szalpal) [reply dali/test/python/test_optical_flow.py]: done

File: dali/test/python/test_optical_flow.py (hunk 2 above)
Comment (szalpal) [reply dali/test/python/test_optical_flow.py]: done

File: dali/test/python/test_optical_flow.py [hunk 3]
```diff
@@ -23,13 +23,14 @@ def __init__(self, batch_size, num_threads, device_id):
...
 
 
 def test_of():
     pipe = OFPipeline(batch_size=batch_size, num_threads=1, device_id=0)
     pipe.build()
     pipe_out = pipe.run()
     frames = pipe_out[0].as_cpu().as_array()
-    myarray=np.loadtxt(image_dir+'decoded_flow_vector.dat')
-    assert (0.9 < np.mean(np.abs(frames[0][0].flatten()-myarray)))
+    myarray = np.loadtxt(image_dir + '/../decoded_flow_vector.dat')
```
Comment (banasraf) [inline dali/test/python/test_optical_flow.py]: Maybe to be consistent, another path created with:
`os.path.join(dali_extra_path, 'db', 'optical_flow', 'slow_preset', 'decoded_flow_vector.dat')`
With it we could get rid of this `/../`

File: dali/test/python/test_optical_flow.py (hunk 1 above)
Comment (klecki) [reply dali/test/python/test_optical_flow.py]: Wouldn't that work:
```suggestion
image_dir = os.path.join(dali_extra_path, "db/optical_flow/slow_preset/two_frames/")
```
?
In any case, I think 
`os.path.join(dali_extra_path, "db", "optical_flow", "slow_preset", "two_frames")`
is worse than 
`dali_extra_path + "/db/optical_flow/slow_preset/two_frames/"` - it's less readable, and "//" is still accepted as "/".
Unless you want to run it on windows, which won't work at all in any foreseeable future.

File: dali/test/python/test_optical_flow.py (hunk 3 above)
Comment (szalpal) [reply dali/test/python/test_optical_flow.py]: done

**Pull request merged**